Environment_Engine/Modify/SetPanelTypeByRegion.cs

```diff
@@ -0,0 +1,104 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.ComponentModel;
+
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Geometry;
+using BH.oM.Environment.Elements;
+using BH.oM.Geometry;
+using BH.oM.Analytical.Elements;
+
+namespace BH.Engine.Environment
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns a list of Environment Panels with panel type set by an IRegion perimeter")] // Not an accurate description
+        [Input("panels", "A collection of Environment Panels to set the type for")]
+        [Input("regions", "A collection of IRegions to set the type by")]
+        [Output("panels", "A collection of modified Environment Panels with the with panel type set by an IRegion perimeter")]
+        public static List<Panel> SetPanelTypeByRegion(List<Panel> panels, List<IRegion> regions)
```

Comment:
Change input to single `Dwelling` object rather than `List<IRegion>` as discussed